example/materialDatabase.js

```diff
@@ -0,0 +1,395 @@
+
+		renderer.toneMapping = value ? THREE.ACESFilmicToneMapping : THREE.NoToneMapping;
+		blitQuad.material.needsUpdate = true;
+
+	} );
+	ptFolder.add( params, 'stableNoise' ).onChange( value => {
+
+		ptRenderer.stableNoise = value;
+
+	} );
+	ptFolder.add( params, 'multipleImportanceSampling' ).onChange( value => {
+
+		ptRenderer.material.setDefine( 'FEATURE_MIS', Number( value ) );
+		ptRenderer.reset();
+
+	} );
+	ptFolder.add( params, 'tiles', 1, 4, 1 ).onChange( value => {
+
+		ptRenderer.tiles.set( value, value );
+
+	} );
+	ptFolder.add( params, 'filterGlossyFactor', 0, 1 ).onChange( () => {
+
+		ptRenderer.reset();
+
+	} );
+	ptFolder.add( params, 'bounces', 1, 30, 1 ).onChange( () => {
+
+		ptRenderer.reset();
+
+	} );
+	ptFolder.add( params, 'resolutionScale', 0.1, 1 ).onChange( () => {
+
+		onResize();
+
+	} );
+
+	const envFolder = gui.addFolder( 'Environment' );
+	envFolder.add( params, 'environmentIntensity', 0, 10 ).onChange( () => {
+
+		ptRenderer.reset();
+
+	} );
+	envFolder.add( params, 'environmentRotation', 0, 2 * Math.PI ).onChange( v => {
+
+		ptRenderer.material.environmentRotation.setFromMatrix4( new THREE.Matrix4().makeRotationY( v ) );
+		ptRenderer.reset();
+
+	} );
+	envFolder.add( params, 'environmentBlur', 0, 1 ).onChange( () => {
+
+		updateEnvBlur();
+
+	} );
+	envFolder.add( params, 'backgroundBlur', 0, 1 ).onChange( () => {
+
+		ptRenderer.reset();
+
+	} );
+
+	animate();
+
+}
+
+function onResize() {
+
+	const w = window.innerWidth;
+	const h = window.innerHeight;
+	const scale = params.resolutionScale;
+	const dpr = window.devicePixelRatio;
+
+	ptRenderer.setSize( w * scale * dpr, h * scale * dpr );
+	ptRenderer.reset();
+
+	renderer.setSize( w, h );
+	renderer.setPixelRatio( window.devicePixelRatio * scale );
+
+	const aspect = w / h;
+
+	perspectiveCamera.aspect = aspect;
+	perspectiveCamera.updateProjectionMatrix();
+
+}
+
+function updateEnvBlur() {
+
+	const blurredTex = envMapGenerator.generate( envMap, params.environmentBlur );
+	ptRenderer.material.envMapInfo.updateFrom( blurredTex );
+	scene.environment = blurredTex;
+	ptRenderer.reset();
+
+}
+
+function applyMaterialInfo( info, material ) {
+
+	material.color.set( 0xffffff );
+	material.transmission = 0.0;
+	material.attenuationDistance = Infinity;
+	material.attenuationColor.set( 0xffffff );
+	material.specularColor.set( 0xffffff );
+	material.metalness = 0.0;
+	material.roughness = 1.0;
+	material.ior = 1.5;
+	material.thickness = 1.0;
+
+	if ( info.specularColor ) material.specularColor.setRGB( ...info.specularColor );
+	if ( 'metalness' in info ) material.metalness = info.metalness;
+	if ( 'roughness' in info ) material.roughness = info.roughness;
+	if ( 'ior' in info ) material.ior = info.ior;
+	if ( 'transmission' in info ) material.transmission = info.transmission;
+
+	if ( material.transmission ) {
+
+		if ( info.color ) material.attenuationColor.setRGB( ...info.color );
+		material.attenuationDistance = 200 / info.density;
```

Comment:
cc @AntonPalmqvist

Do you have any recommendations on references for converting material density to three.js' / glTF "attenuationDistance"?

Reply:
To my knowledge, there's no reliable way to convert physical density to attenuation distance.

The reason I've included density in my database is only for physics simulation purposes, such as rigid body dynamics.

Reply:
Oh gotcha - thanks! So for your renders how are you determining the light attenuation to use for the materials? Is there just a constant attenuation used? Or is it just a surface color that tints all transmitted light?

Reply:
I used Blender Cycles to render the shader balls, with the standard Principled BSDF (Disney). I just set the Base Color to the color of the material, and then I set Transmission to 1. There's no attenuation controls in the shader.

I should use another render engine, with more control, but Blender was the quickest and easiest way to do those initial renders.

Reply:
Oh okay thats good information. I'll leave it at the attenuation method, for now, and maybe revisit later.